### PR TITLE
Add Strided Input test for flex attention

### PR DESCRIPTION
**Pull request by joydddd**
Test strided inputs to the flex_attention HOP. Similar to how inputs are generated in 
https://github.com/pytorch/pytorch/blob/main/benchmarks/transformer/score_mod.py. 

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

### CODE REVIEW

**Comment by joydddd** [comment]
Fix flex attention forward for strided inputs.

File: test/inductor/test_flex_attention.py [hunk 1]
```diff
@@ -438,6 +438,47 @@ def test_builtin_score_mods_different_seqlen(
             S,
             D,
         )
+    
+        
+    @supported_platform
+    @common_utils.parametrize("dtype", test_dtypes)
+    @common_utils.parametrize("score_mod", [_identity, _causal])
+    def test_strided_inputs(self, dtype: torch.dtype, score_mod: Callable):
```
Comment (Chillee) [inline test/inductor/test_flex_attention.py:446]: Could you add some more tests with other permutations of the input matrices?

File: torch/_inductor/kernel/flex_attention.py [hunk 2]
```diff
@@ -194,10 +194,12 @@ def build_subgraph_buffer(
     MATMUL_PRECISION = Q.dtype.element_ty
 
     start_m = tl.program_id(0)
-    off_hz = tl.program_id(1)
+    off_z = tl.program_id(1) // H
+    off_h = tl.program_id(1) % H
```
Comment (drisspg) [inline torch/_inductor/kernel/flex_attention.py:198]: great catch though!

File: test/inductor/test_flex_attention.py (hunk 1 above)
Comment (joydddd) [reply test/inductor/test_flex_attention.py]: Added more strided inputs https://github.com/pytorch/pytorch/pull/128915/commits/947a1e0237bf0d089ec6f716546139d4018d023c

File: test/inductor/test_flex_attention.py [hunk 3]
```diff
@@ -120,6 +120,13 @@ def _trig2(score, b, h, m, n):
 S = 2048
 D = 64
 
+test_input_strides = [
```
Comment (Chillee) [inline test/inductor/test_flex_attention.py:123]: You can move this closer to the actual test definition, since this is only used in one place.

Comment (Chillee) [reply test/inductor/test_flex_attention.py:123]: Also, you might want to write this in terms of B, H, S, and D.

**Review by Chillee — APPROVED** [review]

File: test/inductor/test_flex_attention.py (hunk 3 above)
Comment (joydddd) [reply test/inductor/test_flex_attention.py]: Fixed code style 
 https://github.com/pytorch/pytorch/pull/128915/commits/b345ae6dc34af6393ef88a68dd886ff5f8206b9a

**Review by drisspg — APPROVED** [review]

**Comment by huydhn** [comment]
@pytorchbot revert -m 'Sorry for reverting your change but its tests are failing in trunk, i.e. https://hud.pytorch.org/pytorch/pytorch/commit/41bb81b58279f492e72bd270b3b071dd2953ed8c#26627138290' -c weird

I think this is a land race because your base commit is too old (20 days). It would also explain why there are many unrelated failures in the PR

**Pull request closed without merging**